ctl_main.c

```diff
@@ -1238,6 +1238,58 @@ static int cmd_showport(int argc, char *const *argv)
     return r;
 }
 
+static int cmd_showportparams(int argc, char *const *argv)
+{
+    int r = 0;
+
+    int br_index = get_index(argv[1], "bridge");
+    if(0 > br_index)
+        return br_index;
+
+    int i, count = argc - 3;
+    param_id_t param_id[count];
```

Comment:
this `    param_id_t param_id[count];`  static initialization of array if more C99 specific;

Reply:
any way, we need initialize the array?

Reply:
apologies for the slow reply;
i missed your replies and i am a bit overloaded;

so, typically in older C versions you would allocate a static array of max-values
so, you would add a value to the enum called `PARAM_MAX` and do `param_id_t param_id[PARAM_MAX]`

the difference is that `PARAM_MAX` is a constant and `count` is a run-time variable;
older versions of the C compiler (C90 and below) don't know how to handle `param_id[count]`, so what you would do with these compilers is to allocate a max-array;

the run-time overhead isn't too big to do ` param_id[PARAM_MAX]` vs ` param_id[count]`